planner/python_poetry_planner.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package planner
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pelletier/go-toml/v2"
+)
+
+type PythonPoetryPlanner struct{}
+
+// PythonPoetryPlanner implements interface Planner (compile-time check)
+var _ Planner = (*PythonPoetryPlanner)(nil)
+
+func (g *PythonPoetryPlanner) Name() string {
+	return "PythonPoetryPlanner"
+}
+
+func (g *PythonPoetryPlanner) IsRelevant(srcDir string) bool {
+	poetryLockPath := filepath.Join(srcDir, "poetry.lock")
+	mainPYPath := filepath.Join(srcDir, "main.py")
+	return fileExists(poetryLockPath) && fileExists(mainPYPath)
+}
+
+func (g *PythonPoetryPlanner) GetPlan(srcDir string) *Plan {
+	version := g.PythonVersion(srcDir)
+	return &Plan{
+		Packages: []string{
+			fmt.Sprintf("python%s", version.majorMinorConcatenated()),
+			"poetry",
+		},
+		InstallStage: &Stage{
+			Command: "poetry install --no-dev --no-interaction --no-ansi",
+		},
+		BuildStage: &Stage{
+			Command: "poetry build",
+		},
+		// TODO parse pyproject.toml to get the start command?
+		StartStage: &EntrypointStage{
+			Entrypoint:     "python main.py",
```

Comment:
I'm not sure if we should call it `Entrypoint` because often times I actually want the entrypoint to be `/bin/sh` and the start command to be `yarn start`, so that I can sh into the running container easily and test things out. 

I think separating the concept here would help with that. I'd vote for naming this `Command` and use it as an `Entrypoint` for now. But I see that assumption changing later down the line and using the `Command` as `CMD`. Thoughts?

Reply:
If we keep calling it command I think we should use `CMD` instead. I think calling it `Command` and using it for `ENTRYPOINT` is confusing.